### DISCUSSION TITLE
Trouble with 'end-session-url' config property

### DISCUSSION THREAD

**Discussion by SergeyIntyashev** [trinodb/trino#19490] [Q&A]
I have a problem starting the coordinator with the end-session-url configuration property set. I set the parameter according to the documentation, but the coordinator does not start.

![image](https://github.com/trinodb/trino/assets/52782418/55d1735d-c0bf-4def-8438-438367453912)
![image](https://github.com/trinodb/trino/assets/52782418/442986ec-b260-4765-8eb8-a6f818e02a82)

I use Trino 429 version.

**Comment by Tomme** [comment]
The `http-server.authentication.oauth2.oidc.discovery` property defaults to `true` which means it is retrieving the `end_session_endpoint` values from the Authorization Server, thus you should not set `http-server.authentication.oauth2.end-session-url`

If you set `http-server.authentication.oauth2.oidc.discovery=false` then you can use `http-server.authentication.oauth2.end-session-url` but you will be required to set the following properties as well:
* `http-server.authentication.oauth2.auth-url`
* `http-server.authentication.oauth2.token-url`
* `http-server.authentication.oauth2.jwks-url` 

**I have noticed a potential bug though 👀** , if I set `http-server.authentication.oauth2.oidc.discovery=true` and my Authorization Server does not return a `end_session_endpoint` value you will get the error: `Invalid response from OpenID Metadata endpoint. Missing required "end_session_endpoint" property` - meaning `end_session_endpoint` is a required field, [code](https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/server/security/oauth2/OidcDiscovery.java#L118), which is not helpful since quite a lot of OAuth providers e.g. Google do not provide one: https://accounts.google.com/.well-known/openid-configuration

**Reply by Tomme** [reply]
#19844
